### PR TITLE
unfurling: integrate unfurl prompt with text messages CORE-9476

### PR DESCRIPTION
**Pull request by mmaxim**
@keybase/react-hackers 

Patch does the following:
1.) Adds `UnfurlPromptList` component which displays a list of `UnfurlPrompt` for all the domains in the list.
2.) Adds `unfurlPrompts` to `Types.MessageText` in order to track the unfurl prompts for each sent message.
3.) Hookup listener for `chat.1.NotifyChat.ChatUnfurlPrompt` and dispatch a new action to update the unfurl list for the message in question.
4.) Hookup buttons on `UnfurlPrompt` to hit the RPC for setting unfurl settings on the service. Also dismiss the prompt when this happens.

### CODE REVIEW

File: shared/chat/conversation/messages/unfurl/prompt-list/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          key={p.domain}
+          domain={p.domain}
+          onAlways={p.onAlways}
+          onAccept={p.onAccept}
+          onNotnow={p.onNotnow}
+          onNever={p.onNever}
+        />
+      )
+    }
+    return <React.Fragment>{prompts}</React.Fragment>
```
[the remark this answers is not in the document]
Comment (mmaxim) [reply shared/chat/conversation/messages/unfurl/prompt-list/index.js]: I actually change this in the next PR to be in a box, so maybe I'll just hold off here for that one.

File: shared/chat/conversation/messages/unfurl/prompt-list/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// @flow
+import * as React from 'react'
+import * as Prompt from '../prompt'
```
[the remark this answers is not in the document]
Comment (mmaxim) [reply shared/chat/conversation/messages/unfurl/prompt-list/index.js]: I pulled out the `Props` type too here, so I did it the way I had. Maybe there is a better way to do that.

[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/chat/conversation/messages/unfurl/prompt-list/index.js]: you can do
```
import Prompt, {type whatever} from '../prompt'
```

File: shared/chat/conversation/messages/unfurl/prompt-list/container.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  }
+}
+
+const mapDispatchToProps = (dispatch, {conversationIDKey}: OwnProps) => {
+  const simpleResp = (actionType: any, messageID: Types.MessageID, domain: string) => {
+    dispatch(
+      Chat2Gen.createUnfurlResolvePrompt({
+        conversationIDKey,
+        messageID,
+        domain,
```
[the remark this answers is not in the document]
Comment (mmaxim) [reply shared/chat/conversation/messages/unfurl/prompt-list/container.js]: Both

**Comment by mmaxim** [comment]
@chrisnojima PTAL

File: shared/chat/conversation/messages/unfurl/prompt-list/container.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+const mapStateToProps = (state, {conversationIDKey, ordinal}: OwnProps) => {
+  const message = Constants.getMessage(state, conversationIDKey, ordinal)
+  return {
+    promptDomains: message && message.type === 'text' ? message.unfurlPrompts : noPrompts,
+    messageID: message && message.type === 'text' ? message.id : noMessageID,
+  }
+}
+
+const mapDispatchToProps = (dispatch, {conversationIDKey}: OwnProps) => {
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/unfurl/prompt-list/container.js]: can be 
```
=> ({
...
})
```

File: shared/chat/conversation/messages/unfurl/prompt-list/container.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+          messageID,
+          domain,
+          result,
+        })
+      )
+    },
+  }
+}
+
+const makeRes = (actionType: any, domain?: string) => {
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/unfurl/prompt-list/container.js]: why do you need this?

File: shared/chat/conversation/messages/unfurl/prompt-list/index.stories.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
+// @flow
+import * as React from 'react'
+import {Box} from '../../../../../common-adapters/index'
+import * as Sb from '../../../../../stories/storybook'
+import UnfurlPromptList from '.'
+
+const prompts = [
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/unfurl/prompt-list/index.stories.js]: can be
```
const common = {
  onAlways: Sb.action('onAlways'),
  onAccept: Sb.action('onAccept'),
  onNotnow: Sb.action('onNotnow'),
  onNever: Sb.action('onNever'),
}

const props = {prompts: ['cnn.com', 'github.com', 'wjs.com'].map(domain => ({domain,...common})]}
```

File: shared/chat/conversation/messages/unfurl/prompt-list/container.js (hunk 5 above)
Comment (mmaxim) [reply shared/chat/conversation/messages/unfurl/prompt-list/container.js]: I couldn't figure out how to get Flow to pass without taking `actionType` to `any`. I don't understand the error it gives back so I went with this, which I thought looks ok anyway.

Comment (mmaxim) [reply shared/chat/conversation/messages/unfurl/prompt-list/container.js]: If you could fix that would be great

**Pull request merged**